spec/unit/pops/lookup/context_spec.rb

```diff
@@ -171,13 +171,8 @@ module Lookup
 
       context 'and multiple compilations' do
 
-        before(:each) { Puppet.settings[:environment_timeout] = 'unlimited' }
-        after(:each) do
-          Puppet.settings[:environment_timeout] = 0
-          Puppet.lookup(:environments).clear_all
-        end
-
         it 'will reuse cached_file_data and not call block again' do
+          Puppet.settings[:environment_timeout] = 'unlimited'
```

Comment:
The plot thickens. Setting a debug breakpoint in the cached_file_data makes the test pass on osx.